thrust/testing/discard_iterator.cu

```diff
@@ -13,7 +13,8 @@ void TestDiscardIteratorTraits()
                                                                                thrust::any_system_tag,
                                                                                thrust::random_access_traversal_tag>;
 
-  static_assert(cuda::std::is_same_v<traits::difference_type, ptrdiff_t>);
+  static_assert(cuda::std::is_same_v<traits::difference_type, cuda::_IotaDiffT<cuda::std::ptrdiff_t>>);
+  static_assert(cuda::std::is_same_v<traits::difference_type, long long>);
```

Comment:
Critical: This may be a breaking change on some systems. I don't think we can ship this before CCCL 4.0.